.github/workflows/build_and_test.yml

```diff
@@ -52,7 +52,7 @@ on:
 jobs:
   precondition:
     name: Check changes
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```

Comment:
Here is my theory.

According to the above @HyukjinKwon analysis, this line could be the reason why this PR regenerates the cached images of branch-3.5 and branch-3.4.

`precondition` job is literally changed here although the actual version is the same, Ubuntu 22.04 == Ubuntu Latest. And, it  invokes the following.

https://github.com/apache/spark/blob/714feb7b5aec948c3499ad486dc63ea89a241ffa/.github/workflows/build_and_test.yml#L111-L118

If we look at `Starting job container` stage of `branch-3.5` job, `ghcr.io/apache/apache-spark-ci-image:branch-3.5-8276441035` image has different sha values. It seems to be regenerated.

**BEFORE**: https://github.com/apache/spark/actions/runs/8276441035/job/22645041254

```
Digest: sha256:c97bcef01a653038dba464bc5d947340be4385efa18808b4a3eb813095ea95f9
Status: Downloaded newer image for ghcr.io/apache/apache-spark-ci-image:branch-3.5-8276441035
```

**AFTER**: https://github.com/apache/spark/actions/runs/8276441035/job/22733550856
```
Digest: sha256:a91feba3ada152b513192acb4fa3b2930655e37644f6bad11c30d6c7a6a12383
Status: Downloaded newer image for ghcr.io/apache/apache-spark-ci-image:branch-3.5-8276441035
```

WDTY, @HyukjinKwon and @panbingkun ?

Reply:
Yes, correctly. This only reveals the technical depth, @panbingkun .